app/src/main/java/party_queue/myapplication/SpotifyActivity.java

```diff
@@ -128,31 +129,63 @@ public void onError(Throwable throwable) {
+            case R.id.button:
+                type = "track";
+                break;
+            case R.id.button2:
+                type = "artist";
+                break;
 
+        }
         JsonObjectRequest jsObjRequest = new JsonObjectRequest
-                (Request.Method.GET, "https://api.spotify.com/v1/search?q="+search+"*&type=track&limit=10" , null, new Response.Listener<JSONObject>() {
+                (Request.Method.GET, "https://api.spotify.com/v1/search?q="+search+"*&type="+ type +"&limit="+numResultsToShow , null, new Response.Listener<JSONObject>() {
 
                     @Override
                     public void onResponse(JSONObject response) {
-                        // mTxtDisplay.setText("Response: " + response.toString());
-                        String[] songId = new String[numResultsToShow];
-                        String[] artistnames = new String[numResultsToShow];
-                        String[] songName = new String[numResultsToShow];
-                        try {
-                            for (int i=0; i<10; i++) {
-                                //artist = response.getJSONArray("artists").getJSONObject(0).getString("name");
-                                songId[i] = response.getJSONObject("tracks").getJSONArray("items").getJSONObject(i).getString("id");
-                                artistnames[i] = response.getJSONObject("tracks").getJSONArray("items").getJSONObject(i).getJSONArray("artists").getJSONObject(0).getString("name");
-                                songName[i] = response.getJSONObject("tracks").getJSONArray("items").getJSONObject(i).getString("name");
-                            }
+                        // mTxtDisplay.setText("Response: " + response.toString();
+                        final String[] songId = new String[numResultsToShow];
+                        final String[] artistnames = new String[numResultsToShow];
+                        final String[] songName = new String[numResultsToShow];
+                        String type ="";
+                        switch(button.getId()){
+                            case R.id.button:
+                                type = "track";
+                                try {
+                                    for (int i=0; i<10; i++) {
+                                        //artist = response.getJSONArray("artists").getJSONObject(0).getString("name");
+                                        songId[i] = response.getJSONObject(type+"s").getJSONArray("items").getJSONObject(i).getString("id");
+                                        artistnames[i] = response.getJSONObject(type+"s").getJSONArray("items").getJSONObject(i).getJSONArray("artists").getJSONObject(0).getString("name");
+                                        songName[i] = response.getJSONObject(type+"s").getJSONArray("items").getJSONObject(i).getString("name");
+                                    }
+
+                                } catch (JSONException e) {}
```

Comment:
We should probably at least print a message here in the catch block alerting if this exception has occured.

If this project moves forward past this semester, adding structured error logging/exception handling will be something we will want to do.